utils/dyn-config.c

```diff
@@ -293,6 +296,23 @@ glusterBlockReadConfig(gbConfig *cfg, ssize_t *len)
     *len += m;
   }
 
+  /*
+   * In-case if the editor (vim) follows write to a new file (.swp, .tmp ..)
+   * and move it to actual file name later. There is a window, where we will
+   * encounter one case that the file data is not flushed to the disk, so in
+   * another process(here) when reading it will be empty.
+   *
+   * Let just wait and try again.
```

Comment:
@lxbsz we have handled the emty file case, but in our chat you also mentioned that we see errno = 2, in some cases, can we also handle that part ?

Thanks!

Reply:
@pkalever 
Yeah, this have included this case already.
The errno == 2 case and empty case have the same root cause as above memtioned.

Thanks.

Reply:
This is why I added the "retry: " tag just before the fopen.